l5kit/l5kit/simulation/dataset.py

```diff
@@ -39,6 +40,9 @@ def __init__(self, dataset: EgoDataset, scene_indices: List[int], start_frame: i
             scene_dataset = dataset.get_scene_dataset(scene_idx)
             self.scene_dataset_batch[scene_idx] = scene_dataset
 
+        # keep track of original dataset
+        self.recorded_scene_dataset_batch = deepcopy(self.scene_dataset_batch)
```

Comment:
This can be very expensive operation, do we need to copy everything here or just some part of the `EgoDataset` objects ?

Reply:
we may shave off things later, but for now it's safer to have access to the whole dataset I would say